### PR TITLE
PRSDM-2293: Fix role dropdown filter

### PR DESCRIPTION
**Pull request by leeMarchello**
### Description
When an article has multiple roles assigned to it, the role filtering does not work correctly. Example an article that has 'Engineering' and 'Sales' assigned to it will be hidden when the role filter is set to either of the two roles - It should show the article instead of hiding it. 

### Issue
https://spandigital.atlassian.net/browse/PRSDM-2293

### CODE REVIEW

**Comment by meyerhp** [comment]
@leeMarchello It's still only hiding a section if that section is currently selected. See recording.

https://user-images.githubusercontent.com/48946187/172069789-c5a5847f-2b3b-4446-8be6-eed61cbb7259.mov

**Comment by meyerhp** [comment]
@leeMarchello It's still only hiding a section if that section is currently selected. See recording.

https://user-images.githubusercontent.com/48946187/172069789-c5a5847f-2b3b-4446-8be6-eed61cbb7259.mov

File: src/index.js [hunk 1]
```diff
@@ -30,18 +30,23 @@ window.events = new Subject();
 
 // Role filtering
 $(document).ready(function () {
```
Comment (meyerhp) [inline src/index.js:32]: `$(document).ready` is deprecated, please use `$(function () {`

Comment (meyerhp) [inline src/index.js:32]: `$(document).ready` is deprecated, please use `$(function () {`

**Comment by leeMarchello** [comment]
> @leeMarchello It's still only hiding a section if that section is currently selected. See recording.
> 
>  Screen.Recording.2022-06-05.at.22.40.25.mov

Hey @meyerhp I think we need to have a deeper discussion for that particular issue, there's a bit of a limitation on that because the browser only renders the articles for the currently active section and the jquery script extracts the article id from the article itself and uses that ID to find the corresponding navbar section that it needs to hide (by attaching a hidden class to it). But when you then navigate to a new section the whole website reloads and fetches the markup from the server which means it loses the hidden class it had so when jquery tries to hide it again after the reload it cannot do so. 

Also when you are on a different section only that section is rendered in the browser so there is no way for jquery to traverse the the other sections and determine if it needs to be hidden or not based off the currently selected role filter.

**Review by meyerhp — APPROVED** [review]

**Pull request merged**